login-workflow/src/new-architecture/screens/EulaScreen/EulaScreenBase.tsx

```diff
@@ -0,0 +1,80 @@
+/** eslint-ignore */
+import React, { useCallback, useState } from 'react';
+import { EulaScreenProps } from './types';
+import { WorkflowCard } from '../../components/WorkflowCard';
+import { WorkflowCardActions } from '../../components/WorkflowCard/WorkflowCardActions';
+import { WorkflowCardBody } from '../../components/WorkflowCard/WorkflowCardBody';
+import { WorkflowCardHeader } from '../../components/WorkflowCard/WorkflowCardHeader';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Box from '@mui/material/Box';
+import DOMPurify from 'dompurify';
+
+/**
+ * Component that renders a screen displaying the EULA and requests acceptance via a checkbox.
+ *
+ * @param eulaContent the content to render for the EULA. Can be a plain string or HTML
+ * @param htmlEula true if the EULA should be rendered as HTML
+ * @param checkboxLabel label for the EULA checkbox
+ * @param initialCheckboxValue used to pre-populate the checked/unchecked checkbox when the screen loads
+ * @param checkboxProps used to set checkbox props
+ * @category Component
+ */
+
+export const EulaScreenBase: React.FC<React.PropsWithChildren<EulaScreenProps>> = (props) => {
+    const {
+        onEulaAcceptedChange = (accepted: boolean): boolean => accepted,
+        eulaContent,
+        checkboxLabel,
+        htmlEula,
+        initialCheckboxValue,
+        checkboxProps,
+    } = props;
+
+    const cardBaseProps = props.WorkflowCardBaseProps || {};
+    const headerProps = props.WorkflowCardHeaderProps || {};
+    const actionsProps = props.WorkflowCardActionsProps || {};
+
+    const [eulaAccepted, setEulaAccepted] = useState(onEulaAcceptedChange(initialCheckboxValue) ?? false);
+
+    const handleEulaAcceptedChecked = useCallback(
+        (accepted: boolean) => {
+            setEulaAccepted(onEulaAcceptedChange(accepted));
+        },
+        [onEulaAcceptedChange]
+    );
+
+    return (
+        <WorkflowCard {...cardBaseProps}>
+            <WorkflowCardHeader {...headerProps}></WorkflowCardHeader>
```

Comment:
You can use the WorkflowCardHeader component as self closing element, as we are not passing any children to it.   
```suggestion
            <WorkflowCardHeader {...headerProps} />
```